org-ml.el

```diff
@@ -2338,6 +2338,10 @@ All other arguments follow the same rules as `org-ml-build-table'."
   (->> (-map #'org-ml-build-table-row! row-lists)
        (apply #'org-ml-build-table :tblfm tblfm :post-blank post-blank)))
 
+(defun org-ml-build-org-data (&rest headline-or-sections-nodes)
+  "Return a new org-data node."
+  (apply #'list 'org-data headline-or-sections-nodes))
```

Comment:
This won't work because it's missing the properties plist for the node which should be between the `org-data` and `headline-or-section-nodes`. To make this confusing, the only properties that `org-data` uses ever (I think) are `:begin` and `:end` which are the starting and ending positions of the node in the buffer. Anything built in `org-ml` sets those properties to `nil` because they don't mean anything (for all it knows, there isn't an org-buffer for those properties to refer to).

You can do all this with three built in functions: `org-ml--build-blank-node` (to build the base node), `org-ml--set-properties-nocheck-nil` (to set `:begin/:end` to nil), and `org-ml-set-children` (to set the child nodes from the one argument, which will also automatically check them against the restriction list you modified in the other commit).